changes/metadata.go

```diff
@@ -0,0 +1,162 @@
+package changes
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// SchemaVersion defines the current JSON schema version for persistent data types (Change, Release, ...)
+const SchemaVersion = "1.0"
+
+// Metadata is a representation of the change metadata stored in a .changes directory.
+type Metadata struct {
+	ChangePath string     // ChangePath is the relative path from the current directory to .changes
+	Changes    []*Change  // Changes holds all pending change metadata in the .changes/next-release directory
+	Releases   []*Release // Releases contains all releases in the .changes/releases directory
+}
+
+// LoadMetadata searches from the current directory upwards until it finds a .changes directory, which will be loaded
+// into a new Metadata and returned.
+func LoadMetadata() (*Metadata, error) {
+	path, err := GetChangesPath()
+	if err != nil {
+		return nil, err
+	}
+
+	changes, err := loadChanges(filepath.Join(path, "next-release"))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Metadata{
+		ChangePath: path,
+		Changes:    changes,
+	}, nil
+}
+
+// AddChange adds the given Change to the Metadata's Changes and saves the Change to the next-release directory.
+func (m *Metadata) AddChange(c *Change) error {
+	err := m.SaveChange(c)
+	if err != nil {
+		return err
+	}
+
+	m.Changes = append(m.Changes, c)
+	return nil
+}
+
+// GetChangeById returns the pending Change with the given id.
+func (m *Metadata) GetChangeById(id string) (*Change, error) {
+	_, c, err := m.getChange(id)
+	return c, err
+}
+
+func (m *Metadata) getChange(id string) (int, *Change, error) {
```

Comment:
What was your thought for returning the index value here?

Reply:
RemoveChangeById uses the index to remove the Change from m.Changes.